src/mono/mono/mini/aot-compiler.c

```diff
@@ -8111,7 +8111,14 @@ parse_cpu_features (const gchar *attr)
 		feature = (MonoCPUFeatures) (MONO_CPU_X86_FULL_SSEAVX_COMBINED & ~feature);
 	
 #elif defined(TARGET_ARM64)
-	// TODO: neon, sha1, sha2, asimd, etc...
+	if (!strcmp (attr + prefix, "base"))
+		feature = MONO_CPU_ARM64_BASE;
+	else if (!strcmp (attr + prefix, "neon"))
+		feature = MONO_CPU_ARM64_NEON;
+	else if (!strcmp (attr + prefix, "crypto"))
```

Comment:
Instead of using the crypto name here, could we use "aes" and "sha2" as those are the names that match up to individual instruction set usage on llvm.

Reply:
Here is what LLVM (getHostCPUFeatures) returns for my CPU: X: `neon fp-armv8 crypto crc`
Here is what /proc/cpuinfo says: `fp asimd evtstrm aes pmull sha1 sha2 crc32 cpuid`

Reply:
Which version of LLVM are you working with? I believe this has changed in more recent versions of llvm, and is a total mess. As I've been finishing up my related work with crossgen2 in pr #33274, I've noticed that there doesn't seem to be a compiler switch that enables only sha1, only sha2 which enables both sha1 and sha256. In addition, the behavior of the crypto switch is that it varies based on which baseline architecture is in use. So I believe crypto,armv8.1a has a different meaning than crypto,v8.3a, with regards to which crypto instructions are enabled which is extremely confusing. I believe we should try to make sure these parameters have a direct match to the CPU feature flags, and if that means we need to use different parameter names than llvm, so be it. Looking at the output of `llc-9 -march=aarch64 -mcpu=help`, I see the following relevant features available for the aarch64 architecture.

```
lse                                - Enable ARMv8.1 Large System Extension (LSE) atomic instructions.
neon                               - Enable Advanced SIMD instructions.
sha2                               - Enable SHA1 and SHA256 support.
crc                                - Enable ARMv8 CRC-32 checksum instructions.
```

My proposal is that we match llvm for the `lse`, `neon`, `sha2`, and `crc` parameters, except that we also add another `sha1` which represents access to the `Sha1` managed apis, and `base` which represents access to the `ArmBase` set of managed apis.

Reply:
Got it, remove "crypto" from the PR and left only "crc" since it's the only set my PR implements (+ base one).
We use LLVM 6 (and migrating to LLVM 9)